### PR TITLE
8344420: Remove Security Manager dependencies from javax.security package

### PR DESCRIPTION
**Pull request by seanjmullan**
Now that JEP 486 has been integrated, the `javax.security` package implementation dependencies on `System.getSecurityManager`, `AccessController.doPrivileged` and `AccessControlContext` can be removed.

Most of the changes are straightforward: removal of code calling `System.getSecurityManager()` and unwrapping of code inside `AccessController.doPrivileged`. However, two changes involved slightly more complicated work:

1. `javax.security.auth.login.Configuration` and `javax.security.auth.login.LoginContext` do not need to capture the caller's access control context anymore.
2. The `SecureSet` implementation in `javax.security.auth.Subject` is greatly simplified because it does not do security checks anymore.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8344420](https://bugs.openjdk.org/browse/JDK-8344420): Remove Security Manager dependencies from javax.security package (**Enhancement** - P3)

### Reviewers
 * [Alan Bateman](https://openjdk.org/census#alanb) (@AlanBateman - **Reviewer**)
 * [Roger Riggs](https://openjdk.org/census#rriggs) (@RogerRiggs - **Reviewer**)
 * [Bradford Wetmore](https://openjdk.org/census#wetmore) (@bradfordwetmore - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22368/head:pull/22368` \
`$ git checkout pull/22368`

Update a local copy of the PR: \
`$ git checkout pull/22368` \
`$ git pull https://git.openjdk.org/jdk.git pull/22368/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22368`

View PR using the GUI difftool: \
`$ git pr show -t 22368`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22368.diff">https://git.openjdk.org/jdk/pull/22368.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22368#issuecomment-2498661867)
</details>

### CODE REVIEW

**Review by AlanBateman — APPROVED** [review]
I went through the changes and it looks a good cleanup. I think it would good to have a second reviewer, esp. for Subject and LoginContext, as there is a lot in this cleanup.

File: src/java.base/share/classes/javax/security/auth/Subject.java [hunk 1]
```diff
@@ -1082,31 +1007,15 @@ public int size() {
         }
 
         public Iterator<E> iterator() {
-            final LinkedList<E> list = elements;
+            LinkedList<E> list = elements;
             return new Iterator<>() {
-                final ListIterator<E> i = list.listIterator(0);
+                ListIterator<E> i = list.listIterator(0);
```
[the remark this answers is not in the document]
Comment (seanjmullan) [reply src/java.base/share/classes/javax/security/auth/Subject.java:1012]: I was removing `final` from local variables where I thought it was no longer necessary or did not add much value - this was typically done when using them inside a `doPrivileged` block. I may have gotten a bit too aggressive though. Let me go back and clean this up a bit.

File: src/java.base/share/classes/javax/security/auth/Subject.java [hunk 2]
```diff
@@ -1180,28 +1058,17 @@ public boolean add(E o) {
                 return elements.add(o);
             else {
                 return false;
-        }
+            }
```
[the remark this answers is not in the document]
Comment (seanjmullan) [reply src/java.base/share/classes/javax/security/auth/Subject.java:1061]: Agreed, but there are several `switch` statements in this class which are not indented, so I would prefer to not change them. I indented this one line as it was clearly intended to line up with the `else` block.

**Review by bradfordwetmore — APPROVED** [review]
I only looked over javax.security.cert.  That part LGTM.

**Pull request closed without merging**